[WFLY-8131]:  Aliases in credential stores should be case insensitive.

ehsavoie opened the pull request:
Adding validation step to ensure that  the alias is lower case.
Adding parameter on CredentialStore to allow disabling of this validation in case the store supports case sensitivity.

Jira: https://issues.jboss.org/browse/WFCORE-2556
JBEAP: https://issues.jboss.org/browse/JBEAP-8871
Test update PR : https://github.com/wildfly-security-incubator/wildfly/pull/162

bstansberry commented on elytron/src/main/java/org/wildfly/extension/elytron/CredentialStoreAliasDefinition.java:
```diff
@@ -214,16 +170,17 @@ protected void performRuntime(OperationContext context, ModelNode operation, Res
-            transformOperationAddress(operation);
-            super.execute(context, operation);
+            ModelNode transformedOperation = transformOperation(context, operation);
+            if(!CASE_SENSITIVE.resolveModelAttribute(context, context.readResourceFromRoot(context.getCurrentAddress().getParent()).getModel()).asBoolean()) {
```
context.readResourceFromRoot(....., false) to save cloning children that aren't needed.

bstansberry commented on elytron/src/main/resources/org/wildfly/extension/elytron/LocalDescriptions.properties:
```diff
@@ -1169,6 +1169,7 @@ elytron.server-ssl-context.ssl-session.invalidate=Invalidate the SSLSession (Not
 # Credential Store #
 ####################
 elytron.credential-store=Credential store to keep alias for sensitive information such as passwords for external services.
+elytron.credential-store.case-sensitive=Case sensitivity of the credential store.
```
This needs more explanation.

darranl commented:
Sorry we probably should get the Jira issue moved to WFCORE and the commit messages updated for the new number.  As we used to pull in the subsystem in WildFly we were using WFLY issues to make it easier for QE but now we have moved we should use the correct project.

bstansberry commented on elytron/src/main/java/org/wildfly/extension/elytron/CredentialStoreAliasDefinition.java:
```diff
@@ -214,16 +170,17 @@ protected void performRuntime(OperationContext context, ModelNode operation, Res
-            super.execute(context, operation);
+            ModelNode transformedOperation = transformOperation(context, operation);
+            if(!CASE_SENSITIVE.resolveModelAttribute(context, context.readResourceFromRoot(context.getCurrentAddress().getParent()).getModel()).asBoolean()) {
+                ALIAS_NAME.resolveModelAttribute(context, transformedOperation).asString();
```
A reusable validator is nice, but this isn't a parameter, so the validator is providing a misleading message. I think we're better off providing a precise message, since this kind of constraint on resource names is not common and the reason why case matters isn't obvious.

ehsavoie committed:
[WFCORE-2556]:  Aliases in credential stores should be case insensitive.

Adding validation step to ensure that  the alias is lower case.
Adding parameter on CredentialStore to allow disabling of this validation in case the store supports case sensitivity.

bstansberry commented on elytron/src/main/java/org/wildfly/extension/elytron/CredentialStoreAliasDefinition.java:
```diff
@@ -215,16 +160,19 @@ protected void performRuntime(OperationContext context, ModelNode operation, Res
-            }
+            context.addResource(PathAddress.EMPTY_ADDRESS, resource);
             return resource;
         }
```
This doesn't need to be changed to merge this but I think this entire method override can be removed.

kabir reviewed (approved):
https://issues.jboss.org/browse/JBEAP-8871 is recorded as resolved. Possibly because half the work came from the subsystem PR linked by 8871?
